Update Android ExecuTorch Llama demo app readme docs

cmodi-meta opened the pull request:
Summary:
* Revamp and standardize the readme docs structure for better clarity.
* Added XNNPACK and MTK delegate instructions for Android.
* Added app screenshots for user instruction.
* Added a `docs` directory to host README materials

WuhanMonkey commented on examples/demo-apps/android/LlamaDemo/docs/delegates/mtk_README.md:
Can we use full name mediatek instead of mtk here?

cmodi-meta committed:
Update Android ExecuTorch Llama demo app readme docs (#5364)

Summary:
* Revamp and standardize the readme docs structure for better clarity.
* Added XNNPACK and MTK delegate instructions for Android.
* Added app screenshots for user instruction.
* Added a `docs` directory to host README materials

Pull Request resolved: https://github.com/pytorch/executorch/pull/5364

Differential Revision: D62667589

Pulled By: cmodi-meta

cmodi-meta committed:
`lintrunner -a` resolve
